Dockerfile

```diff
@@ -0,0 +1,40 @@
+### STAGE 1: Build the VueJS app ###
+
+FROM node:lts-alpine as build-stage
+
+WORKDIR /app
+
+COPY package*.json ./
+
+RUN npm install
+
+COPY . .
```

Comment:
Why do not have this at line 7? then you only need one `COPY`

Reply:
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache

Please check the above URL, layer caching is one of the best practices in Docker image.
If we don't separate the 2 COPY(s), whenever we push a new change, Docker builds everything from that point without using cache.

If we separate them, only when we update `package.json` file (which we rarely do), the image is rebuilt from that point.

So it's used to reduce the build time.

Reply:
Good to know!